refactor(daffio): convert header components to standalone

ali-toghiani opened the pull request:
https://github.com/graycoreio/daffodil/issues/3971

## PR Checklist

- [x] Commit message follows our [contributing guidelines](https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit)
- [x] Tests added/updated (for bug fixes/features)
- [ ] Documentation added/updated (for bug fixes/features)

## PR Type
<!-- Select the relevant type(s) -->

- [ ] Bug fix
- [ ] Feature
- [ ] Style update
- [x] Refactor
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Docs
- [ ] Performance
- [ ] Other (please describe)

## Current behavior
<!-- Describe the current behavior or link to relevant issue -->
https://github.com/graycoreio/daffodil/issues/3971

Fixes: #

## New behavior
<!-- Describe the changes -->

## Breaking change?

- [ ] Yes
- [x] No

<!-- If yes, describe impact and migration path -->

## Additional context
<!-- Any other relevant information -->

ali-toghiani committed:
refactor(daffio): convert header/components to standalone

https://github.com/graycoreio/daffodil/issues/3971

xelaint commented on apps/daffio/src/app/core/header/components/header/header.component.ts:
```diff
@@ -4,13 +4,24 @@ import {
   changeDetection: ChangeDetectionStrategy.OnPush,
-  standalone: false,
+  imports: [
+    RouterModule,
```
RouterModule, DaffContainerModule, and DaffioHeaderItemDirective don't need to be imported here since they're not used in this component.

xelaint commented on apps/daffio/src/app/core/nav/docs/docs.component.ts:
```diff
@@ -31,7 +31,7 @@ import { DaffioNavLink } from '../link/type';
   changeDetection: ChangeDetectionStrategy.OnPush,
   imports: [
-    DaffioHeaderComponentModule,
+    DaffioHeaderComponent,
```
DaffioHeaderItemDirective needs to be imported. UI looks like this without it:

<img width="648" height="106" alt="Screen Shot 2025-09-23 at 11 37 23 AM" src="https://github.com/user-attachments/assets/f35d5778-15fc-4ca2-9d6d-0df6e7a5da0c" />

xelaint commented on apps/daffio/src/app/core/nav/marketing/marketing.component.ts:
```diff
@@ -34,7 +34,7 @@ import { DaffioNavLink } from '../link/type';
   changeDetection: ChangeDetectionStrategy.OnPush,
   imports: [
-    DaffioHeaderComponentModule,
+    DaffioHeaderComponent,
```
DaffioHeaderItemDirective needs to be imported. UI looks like this without it:

<img width="919" height="67" alt="Screen Shot 2025-09-23 at 11 36 38 AM" src="https://github.com/user-attachments/assets/62271ccd-83b6-4beb-9e5d-276c0a54ee14" />

ali-toghiani committed:
refactor(daffio): remove unused imports

removed RouterModule & DaffContainerModule

https://github.com/graycoreio/daffodil/issues/3971

ali-toghiani committed:
fix(daffio): import DaffioHeaderItemDirective to docs.component.ts

fixes ui issue caused by conversion of header/header component to standalone

https://github.com/graycoreio/daffodil/issues/3971

ali-toghiani committed:
fix(daffio): import DaffioHeaderItemDirective to marketing.component.ts

fixes ui issue caused by conversion of header/header component to standalone

https://github.com/graycoreio/daffodil/issues/3971

ali-toghiani committed:
refactor(daffio): remove unused DaffioHeaderItemDirective import

https://github.com/graycoreio/daffodil/issues/3971

xelaint reviewed (approved):
LGTM. @ali-toghiani in the future, please allow the reviewer to mark comments as resolved instead of doing it yourself.
